### PR TITLE
feat(web): BOAT-837 - costs - journey 1

### PR DESCRIPTION
**Pull request by james-hill-kc**
## Describe your changes

costs - journey 1

## Issue ticket number and link

BOAT-837

## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [ ] I have performed a self-review of my own code
- [ ] I have double checked this work does not include any hardcoded secrets or passwords
- [ ] I have made corresponding changes to the documentation
- [ ] I have provided details on how I have tested my code
- [ ] I have referenced the ticket number above
- [ ] I have provided a description of the ticket
- [ ] I have included unit tests to cover any testable code changes

### CODE REVIEW

File: appeals/web/src/server/appeals/appeal-documents/appeal-documents.mapper.js [hunk 1]
```diff
@@ -66,13 +73,13 @@ export function documentUploadPage(
...
-		multiple: !documentId,
+		multiple: allowMultipleFiles !== undefined ? allowMultipleFiles : !documentId,
 		documentStage: documentStage,
 		serviceName: documentName || pageHeadingText,
-		pageTitle: 'Upload documents',
+		pageTitle: pageHeadingTextOverride || 'Upload documents',
 		appealShortReference: appealShortReference(appealReference),
 		pageHeadingText: pageHeadingText,
-		documentType: documentType,
+		documentType: _documentType,
```
Comment (kandcdev) [inline appeals/web/src/server/appeals/appeal-documents/appeal-documents.mapper.js:82]: same as above^

File: appeals/web/src/server/appeals/appeal-documents/appeal-documents.mapper.js [hunk 2]
```diff
@@ -49,13 +52,17 @@ export function documentUploadPage(
...
+	documentType
 ) {
 	const isAdditionalDocument = folderPath.split('/')[1] === 'additionalDocuments';
-	const pageHeadingText = mapAddDocumentsPageHeading(isAdditionalDocument, documentId);
+	const pageHeadingText =
+		pageHeadingTextOverride || mapAddDocumentsPageHeading(isAdditionalDocument, documentId);
 	const pathComponents = folderPath.split('/');
 	const documentStage = pathComponents[0];
-	const documentType = pathComponents[1];
+	const _documentType = documentType || pathComponents[1];
```
[the remark this answers is not in the document]
Comment (james-hill-kc) [reply appeals/web/src/server/appeals/appeal-documents/appeal-documents.mapper.js]: good point, that is a much better name - updated!

File: appeals/web/src/server/appeals/appeal-documents/appeal-documents.mapper.js (hunk 1 above)
Comment (james-hill-kc) [reply appeals/web/src/server/appeals/appeal-documents/appeal-documents.mapper.js]: done

**Review by kandcdev — APPROVED** [review]

File: appeals/web/src/server/appeals/appeal-details/__tests__/appeal-details.test.js [hunk 3]
```diff
@@ -784,6 +835,200 @@ describe('appeal-details', () => {
...
+
+				const response = await request.get(`${baseUrl}/${appealId}`);
+
+				expect(response.statusCode).toBe(200);
+
+				const element = parseHtml(response.text);
+
+				expect(element.innerHTML).toMatchSnapshot();
+
+				const lpaCostsDocumentationElement = parseHtml(response.text, {
```
Comment (AlexMoulin) [inline appeals/web/src/server/appeals/appeal-details/__tests__/appeal-details.test.js:894]: Very nice!

File: appeals/web/src/server/appeals/appeal-details/costs/costs.mapper.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+ * @returns {PageContent}
+ */
+export function addDocumentTypePage(appealDetails, documentTypes) {
+	const shortAppealReference = appealShortReference(appealDetails.appealReference);
+
+	/** @type {PageContent} */
+	const pageContent = {
+		title: `What is the document type? - ${shortAppealReference}`,
+		backLinkUrl: `/appeals-service/appeal-details/${appealDetails.appealId}`,
+		heading: 'What is the document type?',
```
Comment (AlexMoulin) [inline appeals/web/src/server/appeals/appeal-details/costs/costs.mapper.js:23]: Are you not missing the preheading with "Appeal ${short appeal reference}?

**Review by AlexMoulin — APPROVED** [review]
Small note but overall good!

File: appeals/web/src/server/appeals/appeal-details/costs/costs.mapper.js (hunk 4 above)
Comment (james-hill-kc) [reply appeals/web/src/server/appeals/appeal-details/costs/costs.mapper.js:23]: It's not displayed on the page in either the mural or the prototype so I think it's been intentionally omitted by design

**Review by kandcdev — APPROVED** [review]

**Pull request merged**